File: 算法/7. 整数反转/Reverse.java
public class Reverse {

    class Solution {
        public int reverse(int x) {
            long n = 0;
            while(x != 0) {
                int a = x % 10;
                n = n * 10 + a;
                x = x / 10;
            }
            return (int) n == n ? (int)n : 0;
        }
    }
    
}

File: 算法/7. 整数反转/Reverse.go
package ___整数反转

func reverse(x int) int {
	result := 0
	for x != 0 {
		a := x % 10
		result = result*10 + a
		x /= 10
	}
	if int(int32(result)) != result {
		return 0
	}
	return result
}
